Dev role name

Jakeasaurus opened the pull request:
# Description
<!-- Description of the changes introduced by this Pull Request (PR). Link to an issue or ticket where possible for more context.-->
Terraform modules to support importing existing AWS IAM resources into Terraform state management.

## What's Changed
•  Updated Role Module - Enables importing existing IAM roles with full configuration support by allowing usage of name OR name_prefix
•  Updated Policy Module - Provides functionality to import existing IAM policies by allowing usage of name or name_prefix

## Benefits
•  Simplifies adoption of existing IAM resources into Terraform workflows

## Issue or Ticket
<!-- Link to the issue or ticket this PR addresses.-->
Fixes #162

## Type of change
<!-- What type of change does your code introduce? -->
- [ ] Bugfix
- [x] New feature
- [ ] Version update

## Breaking Changes
<!-- Does this PR introduce any breaking changes? -->
- [ ] Yes
- [x] No

### Breaking Changes Description
<!-- If yes, describe the breaking changes introduced by this PR. -->


## TODOs
<!-- Complete these tasks prior to requesting a review.-->
- [x] Validate your code matches the style of the project.
- [ ] Update the docs.
- [ ] Validate all tests run successfull, including pre-commit checks.
- [ ] Include release notes and description. This should include both a summary of the changes and any necessary context.

zachreborn reviewed (changes requested):
Let's move this back into the policy and role modules instead of unique modules. See the example code.

Jakeasaurus commented:
Removed _import versions and updated originals. Working run here - https://app.terraform.io/app/SLFCU/workspaces/aws_prod_data/runs/run-dCJwu9cWBZGx2x7t
